sympy/tensor/riemannian_geometry.py

```diff
@@ -0,0 +1,1857 @@
+
+    """
+    # Handling of input vector of arguments - var
+    check_vector_of_arguments(var)
+
+    if isinstance(var, (TensorArray, Arraypy)):
+        var = var.to_list()
+    # Definition of number of variables
+    n = len(var)
+
+    # Handling of a input argument - metric tensor g
+    check_metric_tensor(g)
+
+    if isinstance(g, (Arraypy, TensorArray)):
+        if not (g.start_index[0] == g.start_index[1]):
+            raise ValueError(
+                'The starting indices of metric tensor must be identical')
+        idx_start = g.start_index[0]
+        g_inv = (g.to_matrix()).inv()
+    elif isinstance(g, Matrix):
+        if not g.is_symmetric():
+            raise ValueError('The metric tensor must be symmetric.')
+        idx_start = 0
+        g_inv = g.inv()
+
+    # The definition of diapason changes in an index
+    [n1, n2] = g.shape
+    if not n == n1:
+        raise ValueError(
+            'The rank of the metric tensor does not coincide with the number of variables.')
+
+    indices = range(idx_start, idx_start + n)
+
+    # Creating of output array with new indices
+    Ch = Arraypy([3, n, idx_start])
+
+    # Calculation
+    for i in indices:
+        for j in indices:
+            for k in indices:
+                Ch[i,
+                   j,
+                   k] = sum([g_inv[k - idx_start,
+                                   l - idx_start] * (diff(g[j,
+                                                            l],
+                                                          var[i - idx_start]) + diff(g[i,
+                                                                                       l],
+                                                                                     var[j - idx_start]) - diff(g[i,
+                                                                                                                  j],
+                                                                                                                var[l - idx_start])) / 2 for l in indices])
+
+    # Other variant calculation
+    # christ_1 = christoffel_1(g, var)
+    # for i in indices:
+        # for j in indices:
+            # for k in indices:
+                # Ch[i,
+                # j,
+                # k] = Add(*[g_inv[k,
+                # l] *christ_1[i,
+                # j,
+                # l] for l in indices])
+
+    # Handling of an output array
+    if type_output == str('t') or type_output == Symbol('t'):
+        christoffel_2 = Ch.to_tensor((-1, -1, 1))
+    elif type_output == str('a') or type_output == Symbol('a'):
+        christoffel_2 = Ch
+    else:
+        raise ValueError(
+            "The parameter of type output result must 'a' - Arraypy or 't' and None - TensorArray.")
+
+    # Output
+    return christoffel_2
+
+
+def covar_der(X, g, var, type_output='t'):
+    """Return the covariant derivative the vector field.
+    nabla X[i,j] = diff(X[j],x[i])+Sum_{k}(Gamma2[k,i,j]*X[k])
```

Comment:
LaTeX is enabled in the documentation. You can define the string as `r""" ... ` to avoid the character `\` being escaped.